### PR TITLE
Add Troubleshoot Read and Write latency issue document

### PR DESCRIPTION
**Pull request by TomShawn**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3282
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by TomShawn** [comment]
Some details are to be confirmed with @KASSADAR before this PR can be reviewed.

File: troubleshoot-cpu-issues.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+### CPU load increases
+
+#### Phenomenon
+
+The usage of CPU resources becomes the bottleneck.
+
+#### Possible reasons
+
+* Hotspot issue
+* High overall load. Check the slow queries and expensive queries of TiDB. Optimize the executing queries by adding indexes or executing queries in batches. Another option to handle the issue of high load is to scale out the cluster.
```
Comment (ran-huang) [inline troubleshoot-cpu-issues.md:109]:
```suggestion
* High overall load. Check the slow queries and expensive queries of TiDB. Optimize the executing queries by adding indexes or executing queries in batches. Another solution is to scale out the cluster.
```

File: troubleshoot-cpu-issues.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+## Other causes
+
+### Cluster maintenance
+
+Most of each online cluster has three or five nodes. If the machine to be maintained has the PD component, you need to determine whether the node is the leader or the follower. Disabling a follower has no impact on the cluster operation. Before disabling a leader, you need to switch the leadership. During the leadership change, performance jitter of about 3 seconds will occur.
+
+### Minority of replicas are offline
+
+By default, each TiDB cluster has three replicas, so each Region has three replicas in the cluster. These Regions elect the leader and replicate data through the Raft protocol. The Raft protocol ensures that TiDB can still provide services even when the nodes (that are fewer than half of replicas) fail or are isolated without data loss. For the cluster with three replicas, the failure of one node might cause performance jitter but the usability and correctness in theory are not affected.
```
Comment (ran-huang) [inline troubleshoot-cpu-issues.md:119]:
```suggestion
By default, each TiDB cluster has three replicas, so each Region has three replicas in the cluster. These Regions elect the leader and replicate data through the Raft protocol. The Raft protocol ensures that TiDB can still provide services without data loss even when the nodes (that are fewer than half of replicas) fail or are isolated. For the cluster with three replicas, the failure of one node might cause performance jitter but the usability and correctness in theory are not affected.
```

File: troubleshoot-cpu-issues.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+In general cases, you can first keep their default values (`4` and `256`), observe the resource usage and response speed of the cluster, and then increase the value of `tidb_ddl_reorg_worker_cnt` to increase the concurrency. If no obvious jitter is observed in the monitor, increase the value of `tidb_ddl_reorg_batch_size`. If the columns involved in the index creation are frequently updated, the many resulting conflicts will cause the index creation to fail and be retried.
+
+In addition, you can also set the value of `tidb_ddl_reorg_priority` to `PRIORITY_HIGH` to prioritize the index creation and speed up the process. But in the general OLTP system, it is recommended to keep its default value.
+
+### High GC pressure
+
+The transaction of TiDB adopts the Multi-Version Concurrency Control (MVCC) mechanism. When the newly written data overwrites the old data, the old data is not replaced but stored with the newly written data. Timestamps are used to mark different versions. The task of GC is to clear the obsolete data.
```
Comment (ran-huang) [inline troubleshoot-cpu-issues.md:135]:
```suggestion
The transaction of TiDB adopts the Multi-Version Concurrency Control (MVCC) mechanism. When the newly written data overwrites the old data, the old data is not replaced, and both versions of data are stored. Timestamps are used to mark different versions. The task of GC is to clear the obsolete data.
```

File: troubleshoot-cpu-issues.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+* In the phase of Resolve Locks, a large amount of `scan_lock` requests are created in TiKV, which can be observed in the gRPC-related metrics. These `scan_lock` requests call all Regions.
+* In the phase of Delete Ranges, a few (or no) `unsafe_destroy_range` requests are sent to TiKV, which can be observed in the gRPC-related metrics and the **GC tasks** panel.
+* In the phase of Do GC, each TiKV by default scans the leader Regions on the machine and performs GC to each leader, which be observed in the **GC tasks** panel.
```
Comment (ran-huang) [inline troubleshoot-cpu-issues.md:139]:
```suggestion
* In the phase of Do GC, each TiKV by default scans the leader Regions on the machine and performs GC to each leader, which can be observed in the **GC tasks** panel.
```

File: troubleshoot-cpu-issues.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+* If the query execution plan is output in the slow log, you can directly view the plan. Execute the `select tidb_decode_plan('xxx...')` statement to parse the detailed execution plan.
+* The number of scanned keys in the monitor abnormally increases; in the slow log, the number of `Scan Keys` are large.
+* The SQL execution duration in TiDB is greatly different than that of other databases such as MySQL. You can compare the execution plan of other databases (for example, whether `Join Order` is different).
+
+#### Possible reason
+
+The statistical information is inaccurate.
+
+#### Handling methods
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply troubleshoot-cpu-issues.md:26]:
```suggestion
#### Troubleshooting methods
```

File: troubleshoot-cpu-issues.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+* Bind the execution plan
+    * Modify the application SQL statements and execute `use index` to consistently use the index of the column.
+    * In 3.0 versions, you do not need to modify the application SQL statements. Use `create global binding` to create the binding SQL statement of `force index`.
+    * In 4.0 versions, [SQL Plan Management](/sql-plan-management.md) is supported, which avoids the performance decrease caused by unstable execution plans.
+
+### PD anomalies
+
+#### Phenomenon
+
+There is an abnormal increase of the `wait duration` metric for the PD TSO. This metric represents the duration of waiting for PD to return.
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply troubleshoot-cpu-issues.md:43]: @KASSADAR PTAL
```suggestion
There is an abnormal increase of the `wait duration` metric for the PD TSO. This metric represents the duration of waiting for PD to return requests.
```

[the remark this answers is not in the document]
Comment (TomShawn) [reply troubleshoot-cpu-issues.md:43]: As confirmed with @KASSADAR, the comment above is correct.

**Review by ran-huang — APPROVED** [review]

**Pull request merged**